caluma/caluma_form/jexl.py

```diff
@@ -67,12 +77,59 @@ def _question(self, slug):
             )
         return question
 
+    def _paths_to_question(self, question, doc_form):
+        """Return all paths from the doc form to the given question."""
+        cache = self._cache["containers_hidden_paths"]
+        if question.slug in cache:
+            return cache[question.slug]
```

Comment:
The return value of the function seems to depend on `doc_form` but the caching is only based on `question`. Doesn't this cause issues?

Reply:
There can only be one answer per question in the whole document structure. The only place where that's not true is in table rows. But the "containing question" thing falls down there anyway, as you cannot refer to a specific row anyhow, so the expression cannot be correctly evaluated in that case.

Reply:
Hmm... isn't `doc_form` always the main form? if so, then adding it to the caching keys should not change the cache hit ratio, right?
Or are you calling this with for table row documents as well?

Reply:
Changed the code to include `doc_form` in the cache key